  INFO root [2018-07-15 12:13:35,789] - 
  INFO root [2018-07-15 12:13:35,789] - -----------------------------------------------------------------------------------------


  INFO root [2018-07-15 12:13:35,789] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2018-07-15 12:13:35,789] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/assets_Battle of Gods/;D:/kbengine/assets_Battle of Gods/scripts/;D:/kbengine/assets_Battle of Gods/res/
  INFO root [2018-07-15 12:13:35,789] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2018-07-15 12:13:35,805] - NetworkInterface::initialize(EXTERNAL): bound to all interfaces with default route interface on 10.139.66.192 ( 10.139.66.192:20013 )
  INFO root [2018-07-15 12:13:35,805] - NetworkInterface::initialize(EXTERNAL): address 10.139.66.192:20013, SOMAXCONN=128.
  INFO root [2018-07-15 12:13:35,820] - NetworkInterface::initialize(INTERNAL): bound to all interfaces with default route interface on 10.139.66.192 ( 10.139.66.192:63481 )
  INFO root [2018-07-15 12:13:35,820] - NetworkInterface::initialize(INTERNAL): address 10.139.66.192:63481, SOMAXCONN=128.
  INFO root [2018-07-15 12:13:35,820] - server-configs:
  INFO root [2018-07-15 12:13:35,820] - 	internalAddr : 10.139.66.192:63481
  INFO root [2018-07-15 12:13:35,820] - 	externalAddr : 10.139.66.192:20013
  INFO root [2018-07-15 12:13:35,820] - 	componentID : 9000
  INFO root [2018-07-15 12:13:36,039] - Components::findLogger: found logger, addr:10.139.66.192:63449
  INFO root [2018-07-15 12:13:36,039] - Components::addComponent: reset orderLog, uid=10103!
  INFO root [2018-07-15 12:13:36,039] - Components::addComponent[logger], uid=10103, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2018-07-15 12:13:36,055] - ---- loginapp Version: 2.0.0. ScriptVersion: 0.1.0. Protocol: B20F257E332779D9E1EA7D34638E6146. Config: Debug X86. Built: 16:27:30 Apr 30 2018. AppID: 9000. UID: 10103. PID: 11400 ----
  INFO root [2018-07-15 12:13:36,055] - Server : Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz with 2097151kB RAM
  INFO root [2018-07-15 12:13:36,055] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2018-07-15 12:13:36,055] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2018-07-15 12:13:36,195] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [2018-07-15 12:13:36,195] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [2018-07-15 12:13:36,195] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [2018-07-15 12:13:36,195] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [2018-07-15 12:13:36,195] - Script::install_py_dlls(): PyInit_select
 DEBUG root [2018-07-15 12:13:36,195] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [2018-07-15 12:13:36,195] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [2018-07-15 12:13:36,242] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [2018-07-15 12:13:36,273] - Script::install(): is successfully, Python=(3.4.2 (default, Apr 30 2018, 16:27:27) [MSC v.1900 32 bit (Intel)])!
  INFO root [2018-07-15 12:13:36,273] - Script::install(): KBExtra is successfully!
S_INFO root [2018-07-15 12:13:36,320] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2018-07-15 12:13:36,320] - TelnetServer server is running on port 31000
  INFO root [2018-07-15 12:13:36,695] - ---- loginapp is running ----
 DEBUG root [2018-07-15 12:13:36,695] - Components::process(): Request for the process of identity...
  WARN root [2018-07-15 12:13:43,025] - Forwarding logs to logger[10.139.66.192:63449/0/0/0]...
 ERROR root [2018-07-15 12:17:45,543] - Components::removeComponentByChannel: logger : 2000, Abnormal exit! errorno=10054
  INFO root [2018-07-15 12:17:45,545] - The following logs sent to logger failed:
  WARN root [2018-07-15 12:17:45,546] - ==> [2018-07-15 12:17:45 543] TCPPacketReceiver::processPendingEvents(10.139.66.192:63449): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2018-07-15 12:17:45,546] - ==> [2018-07-15 12:17:45 544] Components::removeComponentByChannel: logger : 2000, Abnormal exit! errorno=10054
  WARN root [2018-07-15 12:17:45,634] - TCPPacketReceiver::processPendingEvents(10.139.66.192:63454): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2018-07-15 12:17:45,635] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit! errorno=10054
  INFO root [2018-07-15 12:17:45,636] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2018-07-15 12:17:45,636] - onLoginAppShutDown()
  INFO root [2018-07-15 12:17:45,637] - Shutdowner::onShutdown: shutting down
  INFO root [2018-07-15 12:17:45,637] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [2018-07-15 12:17:45,637] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [2018-07-15 12:17:45,638] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [2018-07-15 12:17:45,638] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [2018-07-15 12:17:45,638] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [2018-07-15 12:17:45,638] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [2018-07-15 12:17:45,638] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [2018-07-15 12:17:45,639] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [2018-07-15 12:17:45,639] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [2018-07-15 12:17:45,652] - Script::uninstall(): is successfully!
 DEBUG root [2018-07-15 12:17:45,652] - ThreadPool::destroy(): starting size 2.
  WARN root [2018-07-15 12:17:45,953] - ThreadPool::destroy(): waiting for thread(2)[0x18301b0,0x18306f8,], try=1
